phase2/docker-images/toolbox/ansible/aws.yml

```diff
@@ -0,0 +1,179 @@
+      aws_local_hostname: "{{ ec2_describe_instance.instances[0].private_dns_name }}"
+      toolbox_image: "{{ lookup('env','KUBERNETES_ANYWHERE_TOOLBOX_IMAGE') }}"
+
+  - name: Fetch metadata about all other cluster members
+    delegate_to: localhost
+    ec2_remote_facts:
+      region: "{{ ec2_instance_identity.region }}"
+      filters:
+        "tag:KubernetesCluster": "{{ kubernetes_cluster }}"
+    register: ec2_describe_cluster
+
+  - name: Set cluster member list
+    set_fact:
+      kubernetes_cluster_memebers: "{{ ec2_describe_cluster.instances | map(attribute='private_ip_address') | ipv4 | list }}"
+
+  - name: Install all systemd units
+    copy: src="{{ item }}" dest="/etc/systemd/system/" owner="root" group="root" mode="0644"
+    with_fileglob:
+      - "/etc/toolbox/systemd-units/common/*"
+      - "/etc/toolbox/systemd-units/ec2/*"
+      - "/etc/toolbox/systemd-units/with-pki/*"
+
+  - file: path=/etc/kubernetes-anywhere state=directory owner="root" group="root" mode="0755"
+
+  - name: Set hostname for kubelet to use
+    copy: dest="/etc/kubernetes-anywhere/local.env"
+                mode="0644" owner="root" group="root"
+                content="AWS_LOCAL_HOSTNAME=\"{{ aws_local_hostname }}\"\nWEAVE_KNOWN_PEERS=\"{{ kubernetes_cluster_memebers | join(' ') }}\"\n"
+
+  - name: Load new systemd units
+    shell: systemctl daemon-reload
+
+  - name: Enable base systemd units
+    shell: systemctl enable \
+             weave.service \
+             weaveproxy.service \
+             weaveplugin.service \
+             weave-expose.service \
+             weave-network.target
+
+  - name: Enable base systemd units
+    shell: systemctl start weave-network.target
+
+  - name: Start ECR login service
+    when: kubernetes_role == "kubernetes-master"
+    ## as master builds PKI images, it needs to login to ECR first
+    shell: systemctl start ecr-login.service
+
+  - name: Create PKI and push to ECR
+    ## TODO: this currently runs
+    delegate_to: localhost
+    when: kubernetes_role == "kubernetes-master"
+    ## this is to be done only once, so there is no unit for it
+    docker:
+      image: "{{ toolbox_image }}"
+      detach: false
+      volumes:
+        - "/root/.docker/:/root/.docker/"
+        - "/var/run/docker.sock:/docker.sock"
+      command: "create-pki-containers" ## TODO: pass public IP
+
+  - name: Set PKI facts
+    set_fact:
+      master_pki_prefix: "{{ pki_registry_domain }}/{{ kubernetes_cluster }}/master/pki"
+      node_pki_prefix: "{{ pki_registry_domain }}/{{ kubernetes_cluster }}/node/pki"
+
+  - name: Tag master PKI images
+    delegate_to: localhost
+    shell: "docker tag \"kubernetes-anywhere:{{ item }}-pki\" \"{{ master_pki_prefix }}:{{ item }}\""
+    with_items:
+      - apiserver
+      - scheduler
+      - controller-manager
+      - toolbox
+
+  - name: Push master PKI images to ECR
+    delegate_to: localhost
+    when: kubernetes_role == "kubernetes-master"
```

Comment:
Yes, I'm certainly intending to refactor this into separate files, now that's it's pretty much feature-complete.